storage/config.go

```diff
@@ -25,6 +30,12 @@ type Config struct {
 	// as they happen, also known as the write-through caching strategy.
 	// Does not apply if Config.Type is not TypePostgres or TypeSQLite.
 	Caching bool `yaml:"caching,omitempty"`
+
+	// MaximumNumberOfResults is the maximum number of results that an endpoint can have
+	MaximumNumberOfResults int `yaml:"maximumNumberOfResults,omitempty"`
+
+	// MaximumNumberOfEvents is the maximum number of events that an endpoint can have
+	MaximumNumberOfEvents int `yaml:"maximumNumberOfEvents,omitempty"`
```

Comment:
As mentioned in another review comment, `maximumNumberOfResults` and `maximumNumberOfEvents` should be `maximum-number-of-results` and `maximum-number-of-events` respectively.